sdks/python/apache_beam/runners/common.py

```diff
@@ -333,6 +336,7 @@ def is_splittable_dofn(self):
     return self.get_restriction_provider() is not None
 
   def get_restriction_coder(self):
+    # type: () -> Optional[TupleCoder]
```

Comment:
A slight refactor here will avoid introducing a new mypy error:

```python
  def get_restriction_coder(self):
    # type: () -> Optional[TupleCoder]
    """Get coder for a restriction when processing an SDF. """
    if self.is_splittable_dofn():
      return TupleCoder([
          (self.get_restriction_provider().restriction_coder()),
          (self.get_watermark_estimator_provider().estimator_state_coder())
      ])
    else:
      return None
```

This avoids having to declare  the `restriction_coder` variable as `Optional[TupleCoder]`.

Reply:
Done. I think the return type should still be `Optional[TupleCoder]` given that it also returns `None`.

Reply:
correct.  my comment about avoiding the declaration of  `Optional[TupleCoder]` refers to the variable (which my edit does away with), not the the return type.